cmd/snap-preseed/preseed_linux.go

```diff
@@ -83,24 +83,27 @@ var systemSnapFromSeed = func(rootDir string) (string, error) {
 		return "", err
 	}
 
-	// TODO: handle core18, snapd snap.
+	var required string
 	if seed.UsesSnapdSnap() {
-		return "", fmt.Errorf("preseeding with snapd snap is not supported yet")
+		required = "snapd"
```

Comment:
we should probably still fail here if the model is not classic ? that needs its own tests and need to deal with wrappers/core18.go, no ?

Reply:
Hmm you're right,  and I missed wrappers for core18, definately needs a close look, probably needs extra checks in the spread test too. Thanks

Reply:
to be clear the code in wrappers/core18.go is a nop for classic afair

Reply:
Indeed, it's a noop.

Reply:
> we should probably still fail here if the model is not classic ? that needs its own tests and need to deal with wrappers/core18.go, no ?

Added error on non-classic + test.